ufl/algorithms/compute_form_data.py

```diff
@@ -60,8 +60,9 @@ def _compute_element_mapping(form):
     #   worked around to drop this requirement
 
     # Extract all elements and include subelements of mixed elements
-    elements = [obj.ufl_element() for obj in chain(form.arguments(),
-                                                   form.coefficients())]
+    elements_a = tuple(a.ufl_function_space().ufl_element() for a in form.arguments())
+    elements_c = tuple(c.ufl_element() for c in form.coefficients())
```

Comment:
coefficients and arguments both have function spaces attached, so can both these iterations be merged?